packages/server/modules/workspaces/services/regions.ts

```diff
@@ -0,0 +1,39 @@
+import { GetRegions } from '@/modules/multiregion/domain/operations'
+import {
+  AssignRegion,
+  GetAvailableRegions,
+  UpsertRegionAssignment,
+  DeleteAllRegionAssignments
+} from '@/modules/workspaces/domain/operations'
+import { WorkspaceRegionAssignmentError } from '@/modules/workspaces/errors/regions'
+
+export const getAvailableRegionsFactory =
+  (deps: { getRegions: GetRegions }): GetAvailableRegions =>
+  async () => {
+    // TODO: Gatekeeper checks here? For now just returning all server regions
```

Comment:
Yup, lets inject, the ` canWorkspaceUseRegions` function, and if it returns false, lets throw.

Reply:
@gjedlicska Ok, but there also has to be a way to get the specific regions accessible to the workspace. Some can be already bought, others may not be.

Reply:
for now, that doesn't matter. You are not buying access on a per region basis

Reply:
@gjedlicska it does matter in the context of this upcoming task https://linear.app/speckle/issue/WEB-1837/list-available-regions-on-the-server-in-workspace-settings

Reply:
@gjedlicska Also, 1) I'm not able to find "canWorkspaceUseRegions" anywhere in the codebase 2) Why do you think this should throw instead of returning an empty array?